src/ProgressOnderwijsUtils/Data/SqlCommandTracer.cs

```diff
@@ -68,10 +68,16 @@ public SqlTraceEvent[] ListAllCommands()
                 }
             }
 
+            const int maxEventCount = 1000;
+
             public void RegisterEvent(string commandText, TimeSpan duration)
             {
                 lock (allqueries) {
-                    allqueries.Add(new SqlTraceEvent { EventContent = commandText, Duration = duration, CumulativeElapsedTime = ElapsedTime.Elapsed });
+                    if (allqueries.Count < maxEventCount) {
+                        allqueries.Add(new SqlTraceEvent { EventContent = commandText, Duration = duration, CumulativeElapsedTime = ElapsedTime.Elapsed });

```

Comment:
Omdat de lengte van events nogaal kan verschillen, zou het evt. niet gek zijn om de afkapping op basis van totale content length te doen, niet op basis van aantal; dat matched beter met geheugengebruik.

Reply:
Dat vinden we belangrijk?

Reply:
Naja, beetje raison d'être van deze PR, toch?